### PR TITLE
Multi fixture file

### PR DESCRIPTION
**Pull request by nkohen**
Implements a framework for having as many different fixtures used as needed in a single test file.

In order to use this, one must `override type FIxtureParam = ChainFixture` and then tag each test with the correct `FixtureTag` and begin each test with `inFixtured { case ExpectedFixture(...) =>`

Optionally, one can also `override def defaultTag` if the plurality of tests use some fixture other than `Empty`.

I have not implemented this but we could make using this framework mandatory by adding `override type FixtureParam = ChainFixture` in `ChainUnitTest.scala`. Doing this would get rid of all casting and all `with...` defs other than `withChainFixture`. On the other hand it would add a tiny bit to test files that are only using one fixture. Thoughts?

### CODE REVIEW

File: chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala [hunk 1]
```diff
@@ -59,6 +59,150 @@ trait ChainUnitTest
...
+        case GenisisChainHandler.name     => GenisisChainHandler
+        case BitcoindZmqChainHandlerWithBlock.name =>
+          BitcoindZmqChainHandlerWithBlock
+        case _: String =>
+          throw new IllegalArgumentException(s"$tag is not a valid tag")
+      }
+    }
+  }
+
+  def defaultTag: FixtureTag = FixtureTag.Empty
```
Comment (Christewart) [inline chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala]: should be val

File: chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala [hunk 2]
```diff
@@ -59,6 +59,150 @@ trait ChainUnitTest
...
+      val destroyedF =
+        fixtureF.flatMap(fixture => ChainFixture.destroy(fixture))
+
+      destroyedF.map(_ => outcome)
+    }
+
+    new FutureOutcome(fixtureTakeDownF)
+  }
+
+  final class SugaryItVerbStringTaggedAs(
```
Comment (Christewart) [inline chain-test/src/test/scala/org/bitcoins/chain/util/ChainUnitTest.scala:184]: Can we get a comment on the methods and on this class? I have no idea what it is meant to do. This seems to be where the test case actually gets run?
